lib/fs.js

```diff
@@ -283,35 +190,41 @@ fs.access = function(path, mode, callback) {
     throw new errors.TypeError('ERR_INVALID_CALLBACK');
   }
 
-  if (handleError((path = getPathFromURL(path)), callback))
-    return;
+  handleError(path = getPathFromURL(path));
+  nullCheck(path);
 
-  if (!nullCheck(path, callback))
-    return;
+  if (typeof path !== 'string' && !Buffer.isBuffer(path)) {
```

Comment:
We actually accept all TypedArrays right now if I am not mistaken. Did you want to limit it to buffers?

Reply:
I'll double check